uber/models.py

```diff
@@ -1625,6 +1817,464 @@ def must_contact(self):
+            consisting entirely of whitespace.
+        discount (int): The discount amount that should be applied to the
+            purchase price of a badge. The interpretation of this value
+            depends on the value of `discount_type`. In any case, a value of
+            0 equates to a full discount, i.e. a free badge.
+        discount_display (str): A human readable description of the discount.
+        discount_type (int): The type of discount this promo code will apply.
+            Valid values are:
+
+            * 0 `FIXED_DISCOUNT`: `discount` is interpreted as a fixed
+                dollar amount by which the badge price should be reduced. If
+                `discount` is 49 and the badge price is normally $100, then
+                the discounted badge price would be $51.
+
+            * 1 `FIXED_PRICE`: `discount` is interpreted as the actual badge
+                price. If `discount` is 49, then the discounted badge price
+                would be $49.
+
+            * 2 `PERCENT_DISCOUNT`: `discount` is interpreted as a percentage
+                by which the badge price should be reduced. If `discount` is
+                20 and the badge price is normally $50, then the discounted
+                badge price would $40 ($50 reduced by 20%). If `discount` is
+                100, then the price would be 100% off, i.e. a free badge.
+
+        expiration_date (datetime): The date & time upon which this promo code
+            expires. An expired promo code may no longer be used to receive
+            discounted badges.
+        is_expired (bool): True if this promo code is expired, False otherwise.
+        is_unlimited (bool): True if this promo code may be used an unlimited
+            number of times, False otherwise.
+        is_valid (bool): True if this promo code is still valid and may be
+            used again, False otherwise.
+        normalized_code (str): A normalized version of `code` suitable for
+            database queries. Normalization converts `code` to all lowercase
+            and removes dashes ("-").
+        used_by (list): List of attendees that have used this promo code.
+            Note:
+                This property is declared as a backref in the Attendee class.
+        uses_allowed (int): The total number of times this promo code may be
+            used. A value of None means this promo code may be used an
+            unlimited number of times.
+        uses_allowed_display (int): A human readable description of
+            uses_allowed.
+        uses_count (int): The number of times this promo code has already
+            been used.
+        uses_count_display (int): A human readable description of uses_count.
+        uses_remaining (int): Remaining number of times this promo code may
+            be used.
+        uses_remaining_display (int): A human readable description of
+            uses_remaining.
+    """
+
+    FIXED_DISCOUNT = 0
```

Comment:
Defining the config-type variables here is out of style re: the rest of the codebase. Either these should go in configspec.ini and config.py or we should have a conversation about moving to this style of config defining.

FWIW, I'd support having things like `AMBIGUOUS_CHARS` and `UNAMBIGUOUS_CHARS` here, but I think with the other vars you're duplicating work, and those belong better under `[enums]` in a `[[discount]]` section.